piccolo/query/methods/update.py

```diff
@@ -32,15 +33,27 @@ def where(self, where: Combinable) -> Update:
 
     def validate(self):
         if len(self.values_delegate._values) == 0:
-            raise ValueError(
-                "No values were specified to update - please use .values"
-            )
+            raise ValueError("No values were specified to update - please use .values")
 
         for column, value in self.values_delegate._values.items():
             if len(column._meta.call_chain) > 0:
-                raise ValueError(
-                    "Related values can't be updated via an update"
-                )
+                raise ValueError("Related values can't be updated via an update")
+
+    async def run_pre_functions(self):
+        for function in self.table._meta.pre_update:
+            function()
```

Comment:
@coder3112 This is more of a question, shouldn't these be awaited?

Reply:
Good Catch! I just ran it and yes you are right. A couple changes need to be made so i guess I'll close this.